src/Enigma2.cpp

```diff
@@ -1774,51 +1800,41 @@ PVR_ERROR Enigma2::GetDriveSpace(long long *iTotal, long long *iUsed)
     XMLUtils::GetString(hddNode, "e2capacity", capacity);
     XMLUtils::GetString(hddNode, "e2free", freeSpace);
 
-    std::regex regexGB ("^.* GB");
-    std::regex regexMB ("^.* MB");
-    std::regex regexReplaceGB (" GB");
-    std::regex regexReplaceMB (" MB");
-    std::string replaceWith = "";
+    totalKb += GetKbFromString(capacity);
+    freeKb += GetKbFromString(freeSpace);
+  }
 
-    if (regex_match(capacity, regexGB))
-    {
-      //It's GB
-      double gigabytesSize = atof(regex_replace(capacity, regexReplaceGB, replaceWith).c_str());
+  *iTotal = totalKb;
+  *iUsed = totalKb - freeKb;
 
-      total += (long long)(gigabytesSize * 1024 * 1024);
-    }
-    else
-    {
-      //It's MB
-      if (regex_match(capacity, regexMB))
-      {
-        double megabytesSize = atof(regex_replace(capacity, regexReplaceMB, replaceWith).c_str());
+  Logger::Log(LEVEL_INFO, "GetDriveSpace Total: %u, Used %u", *iTotal, *iUsed);
 
-        total += (long long)(megabytesSize * 1024);
-      }
-    }
+  return PVR_ERROR_NO_ERROR;
+}
 
-    if (regex_match(freeSpace, regexGB))
-    {
-      double gigabytesSize = atof(regex_replace(freeSpace, regexReplaceGB, replaceWith).c_str());
+long long Enigma2::GetKbFromString(const std::string &stringInMbGbTb) const
+{
+  long long sizeInKb = 0;
 
-      used += (long long)(gigabytesSize * 1024 * 1024);
-    }
-    else
+  std::vector<std::string> sizes = {"MB", "GB", "TB"};
+  long multiplier = 1024 * 1024;
+  std::string replaceWith = "";
+  for (auto& size : sizes)
```

Comment:
As `size` is not a complex type, but just `std::string`, II would not use `auto` here. But ofc this is just personal taste.

Bit it should read `const --choose-your-type-&`. ;-)

Reply:
Done

Reply:
You prefer auto over real type (here: std::string) even for non.complex types?